File: task_equivalents/python/data_manipulation_preprocessing/02_merge_aggregate_reshape.py
# Extracted from Ch06.py (Q9 preprocessing/model setup)

import numpy as np
import pandas as pd
from sklearn.model_selection import train_test_split

College = pd.read_csv('data/csv/College.csv')
school_name = College.iloc[:, 0]
College = College.rename(index=lambda i: school_name[i])
College.drop(College.columns[[0]], axis=1, inplace=True)

College['Private01'] = np.where(College.Private == 'Yes', 1, 0)
College.drop(['Private'], axis=1, inplace=True)

X = College.drop(['Apps'], axis=1).astype('float64')
Y = College['Apps']
xtrain, xtest, ytrain, ytest = train_test_split(X, Y, test_size=0.5, random_state=1)

File: task_equivalents/r/data_manipulation_preprocessing/02_merge_aggregate_reshape.R
# Extracted from Ch06.R (Q9 preprocessing/model matrix)

College = read.csv('data/csv/College.csv', header=TRUE)
rownames(College) = College[,1]
College = College[,-1]

College$Private01 = ifelse(College$Private == 'Yes', 1, 0)
College$Private = NULL

X = as.data.frame(lapply(subset(College, select=-Apps), as.numeric))
Y = College$Apps

set.seed(1)
train    = sample(1:nrow(College), nrow(College)/2)
test     = setdiff(1:nrow(College), train)

xtrain = X[train,]
xtest  = X[test,]
ytrain = Y[train]
ytest  = Y[test]

# Previous version
#
# College = read.csv('data/csv/College.csv', header=TRUE)
# rownames(College) = College[,1]
# College = College[,-1]
#
# set.seed(1)
# train    = sample(1:nrow(College), nrow(College)/2)
# test     = (-train)
# df.train = College[train,]
# df.test  = College[test,]
#
# x.train = model.matrix(Apps~., df.train)[,-1]
# y.train = df.train$Apps
# x.test  = model.matrix(Apps~., df.test)[,-1]
# y.test  = df.test$Apps
